### DISCUSSION TITLE
Outline/border on text.

### DISCUSSION THREAD

**Discussion by Pucksnok** [missionpinball/missionpinball#29] [Q&A]
![image](https://user-images.githubusercontent.com/84498221/202929045-6a7df3a7-248b-4a96-b108-af7291f4cdee.png)

Hello.
Is there a easy way to outline/border/or_whatever_its_called text in MPF/Kivy?
I do it by doing the text five times (1 time my text, 4 times black text behind) but it isn't optimal and not pretty at all, see picture.

Cheers.

/Gill

**Comment by qcapen** [comment]
You can use [Bitmap Fonts](https://docs.missionpinball.org/en/latest/displays/widgets/bitmap_fonts.html) (see also [bitmap_font](https://docs.missionpinball.org/en/latest/config/bitmap_fonts.html). You'll have to create the font yourself (usually from a true type font), but there are some tools to assist with this process (see the docs or do a search on google).

Quinn

**Reply by Pucksnok** [reply]
Thx, I will check it out. It could fix the scoring outline but I'm hesitant if I want to use that solution with different fonts, font_sizes etc. Since it's bitmaps I (guess) it may look bad when I use different sizes/scaling.

Will try tomorrow. Again, thx.

**Reply by qcapen** [reply]
Unfortunately it’s not a great solution if you need many different font sizes as scaling definitely does not look good, but if you only need a few sizes & typefaces you can get some nice effects.

**Comment by colemanomartin** [comment]
https://docs.missionpinball.org/en/stable/displays/widgets/display/effects.html?highlight=glow#settings-for-glow-effect
Have you tried using the glow effect and setting the gain to 0 with a background color of black?

**Comment by JAMESCARDONA-PINBALL** [comment]
I use bitmap fonts and they work great for all sorts of text effects. Unfortunately they do not scale so you will need to build exact copies of what you want displayed. For example, I have about 5 different versions of the same bitmap font at 5 different sizes. Bitmaps are pictures really so this is what you have to do, they are not scalable like true type fonts.

The nice thing with this is you can do other weird stuff like put actual pictures in the bitmap font instead of letters.
